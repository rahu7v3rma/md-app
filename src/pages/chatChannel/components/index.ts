import ChatHeader from './chatHeader';
import DateHeader from './dateHeader';
import MessageContent from './messageContent';
import MessageInput from './messageInput';
import MessageQuoted from './messageQuoted';

export { ChatHeader, DateHeader, MessageContent, MessageInput, MessageQuoted };
