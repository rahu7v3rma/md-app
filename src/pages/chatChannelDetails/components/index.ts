import ChannelMemberItem from './channelMemberItem';
import Header from './header';
import Title from './title';

export { ChannelMemberItem, Header, Title };
