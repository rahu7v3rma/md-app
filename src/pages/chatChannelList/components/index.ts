import ChannelList from './channelList';
import NoChatAvailable from './noChatAvailable';

export { ChannelList, NoChatAvailable };
