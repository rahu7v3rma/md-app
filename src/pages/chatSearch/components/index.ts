import ChatSearchHeader from './chatSearchHeader';
import UserItemList from './userItemList';

export { ChatSearchHeader, UserItemList };
