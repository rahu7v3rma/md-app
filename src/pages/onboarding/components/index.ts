import Pagination from './pagination';
import Slide from './slide';

export { Pagination, Slide };
