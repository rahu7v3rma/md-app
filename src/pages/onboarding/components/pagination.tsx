import React, { FunctionComponent } from 'react';
import { StyleSheet, View } from 'react-native';

import { Colors } from '@/theme/colors';

import PaginationDot from './paginationDot';

type Props = {
    activePage: number;
    totalPages: number;
    onShowPage: (pageIndex: number) => void;
    testID?: string;
};

const Pagination: FunctionComponent<Props> = ({
    activePage,
    totalPages,
    onShowPage,
    testID
}: Props) => {
    return (
        <View testID={testID} pointerEvents={'box-none'} style={styles.root}>
            {Array.from({ length: totalPages }, (_page, idx) => (
                <PaginationDot
                    key={`pagination-dot-${idx}`}
                    dotStyle={styles.dot}
                    active={idx === activePage}
                    activeAnimatedStyle={{
                        color: Colors.pagination.active_dot,
                        transform: {
                            rotationDegrees: -45
                        }
                    }}
                    inactiveAnimatedStyle={{
                        color: Colors.pagination.inactive_dot,
                        transform: {
                            rotationDegrees: 0
                        }
                    }}
                    onPress={() => onShowPage(idx)}
                />
            ))}
        </View>
    );
};

const styles = StyleSheet.create({
    root: {
        alignItems: 'center',
        justifyContent: 'center',
        paddingHorizontal: 20,
        paddingVertical: 30,
        flexDirection: 'row'
    },
    dot: {
        width: 8,
        height: 8,
        borderRadius: 2
    }
});

export default Pagination;
