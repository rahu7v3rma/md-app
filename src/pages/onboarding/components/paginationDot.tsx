import React, { FunctionComponent } from 'react';
import {
    StyleProp,
    StyleSheet,
    TouchableOpacity,
    ViewStyle
} from 'react-native';
import Animated, {
    interpolate,
    interpolateColor,
    useAnimatedStyle,
    useDerivedValue,
    withTiming
} from 'react-native-reanimated';

type PagingAnimatedStyle = {
    transform?: { rotationDegrees?: number };
    color?: string;
};

type Props = {
    dotContainerStyle?: StyleProp<ViewStyle>;
    dotStyle?: StyleProp<ViewStyle>;
    active: boolean;
    activeAnimatedStyle?: PagingAnimatedStyle;
    inactiveAnimatedStyle?: PagingAnimatedStyle;
    onPress?: () => void;
};

const PaginationDot: FunctionComponent<Props> = ({
    dotContainerStyle,
    dotStyle,
    active,
    activeAnimatedStyle,
    inactiveAnimatedStyle,
    onPress
}: Props) => {
    const animationProgress = useDerivedValue(() => {
        return withTiming(active ? 1 : 0);
    });

    const animatedBackgroundColor = useAnimatedStyle(() => {
        const backgroundColor = interpolateColor(
            animationProgress.value,
            [0, 1],
            [
                inactiveAnimatedStyle?.color || '#fff',
                activeAnimatedStyle?.color || '#000'
            ]
        );

        return {
            backgroundColor
        };
    });

    const animatedTransform = useAnimatedStyle(() => {
        const rotation = interpolate(
            animationProgress.value,
            [0, 1],
            [
                inactiveAnimatedStyle?.transform?.rotationDegrees || 0,
                activeAnimatedStyle?.transform?.rotationDegrees || 0
            ]
        );

        return {
            transform: [
                {
                    rotate: `${rotation}deg`
                }
            ]
        };
    });

    return (
        <TouchableOpacity
            accessible={false}
            style={[styles.root, dotContainerStyle]}
            onPress={onPress}
        >
            <Animated.View
                style={[dotStyle, animatedBackgroundColor, animatedTransform]}
            />
        </TouchableOpacity>
    );
};

const styles = StyleSheet.create({
    root: {
        alignItems: 'center',
        justifyContent: 'center',
        marginHorizontal: 8
    }
});

export default PaginationDot;
