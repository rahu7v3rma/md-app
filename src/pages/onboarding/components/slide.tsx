import React, { FunctionComponent } from 'react';
import { Image, StyleSheet, View } from 'react-native';
import { SvgXml } from 'react-native-svg';

import wolfThingPng from '@/assets/images/wolfThinfPng.png';
import wolfDance from '@/assets/svg/wolfdance.svg';
import wolfNote from '@/assets/svg/wolfNote.svg';
import { Text } from '@/shared';
import { Size } from '@/shared/text';
import { Colors } from '@/theme/colors';

type Props = {
    index: number;
};

const Slide: FunctionComponent<Props> = ({ index }: Props) => {
    return (
        <View style={styles.slide}>
            <View style={styles.content}>
                <View style={styles.iconContainer}>
                    {index === 0 && <SvgXml xml={wolfNote} />}
                    {index === 1 && <Image source={wolfThingPng} />}
                    {index === 2 && <SvgXml xml={wolfDance} />}
                </View>
                {index === 0 && (
                    <>
                        <Text
                            style={styles.title}
                            size={Size.XLarge}
                            fontWeight="bold"
                            color={Colors.text.black_gray}
                        >
                            Create and track your daily habits
                        </Text>
                        <Text
                            style={styles.text}
                            size={Size.XSmall}
                            color={Colors.text.black}
                            fontWeight="400"
                        >
                            Build new habits and keep track of them every day.
                            Share them with your coach to stay on track with
                            your goals
                        </Text>
                    </>
                )}
                {index === 1 && (
                    <>
                        <Text
                            style={styles.titleSecond}
                            size={Size.XLarge}
                            fontWeight="bold"
                            color={Colors.text.black_gray}
                        >
                            Chat with your coach to stay accountable
                        </Text>
                        <Text
                            style={styles.text}
                            size={Size.XSmall}
                            color={Colors.text.black}
                            fontWeight="400"
                        >
                            Communicate with your coach frequently to share your
                            experience and ask questions in real-time
                        </Text>
                    </>
                )}
                {index === 2 && (
                    <>
                        <Text
                            style={styles.title}
                            size={Size.XLarge}
                            testID="titleTextSlide2"
                            fontWeight="bold"
                            color={Colors.text.black_gray}
                        >
                            Learn powerful tips to transform your health
                        </Text>
                        <Text
                            style={styles.text}
                            size={Size.XSmall}
                            color={Colors.text.black}
                            fontWeight="400"
                        >
                            Discover simple and powerful insights that can
                            positively impact your metabolic health
                        </Text>
                    </>
                )}
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    slide: {
        flex: 1,
        marginTop: -400,
        justifyContent: 'center',
        alignItems: 'center'
    },
    content: {
        position: 'absolute',
        alignSelf: 'center',
        width: '80%',
        top: '43%'
    },
    iconContainer: {
        alignItems: 'center'
    },
    title: {
        textAlign: 'center',
        textShadowColor: 'rgba(255, 255, 255, 0.7)',
        textShadowRadius: 6,
        marginBottom: 16
    },
    titleSecond: {
        marginTop: 23,
        textAlign: 'center',
        textShadowColor: 'rgba(255, 255, 255, 0.7)',
        textShadowRadius: 6,
        marginBottom: 16
    },
    text: {
        textAlign: 'center',
        textShadowColor: 'rgba(255, 255, 255, 0.7)',
        textShadowRadius: 6,
        paddingBottom: 30
    }
});

export default Slide;
