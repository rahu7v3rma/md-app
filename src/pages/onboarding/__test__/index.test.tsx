import { NavigationContainer } from '@react-navigation/native';
import React from 'react';
import 'react-native';
import { Provider } from 'react-redux';
import renderer, { act, ReactTestRenderer } from 'react-test-renderer';

import { mockNavigate } from '@/jestSetup';
import Onboarding from '@/pages/onboarding';
import store from '@/store';
import { Colors } from '@/theme/colors';

import { Slide } from '../components';

let tree: ReactTestRenderer;
describe('Onboarding', () => {
    beforeAll(async () => {
        await act(() => {
            tree = renderer.create(
                <NavigationContainer>
                    <Provider store={store}>
                        <Onboarding />
                    </Provider>
                </NavigationContainer>
            );
        });
    });
    it('snapshot onboarding', async () => {
        expect(tree).toMatchSnapshot();
    });
    it('renders without crashing.', async () => {
        expect(tree.root).toBeTruthy();
    });
    it('updates active slide when carousel is scrolled.', async () => {
        const pagination = tree.root.findByProps({
            testID: 'pagination'
        }).props;
        expect(pagination.activePage).toBe(0);
        const carosel = tree.root.findByProps({
            testID: 'carosel'
        }).props;
        await act(() => carosel.onProgressChange(1, 1));
        const paginationUpdate = tree.root.findByProps({
            testID: 'pagination'
        }).props;
        expect(paginationUpdate.activePage).toBe(1);
    });
    it('displays correct number of dots in pagination.', async () => {
        const pagination = tree.root.findByProps({
            testID: 'pagination'
        }).props;
        expect(pagination.totalPages).toBe(3);
    });
    it('updates pagination when the active slide changes.', async () => {
        const pagination = tree.root.findByProps({
            testID: 'pagination'
        }).props;
        expect(pagination.activePage).toBe(1);
        const carosel = tree.root.findByProps({
            testID: 'carosel'
        }).props;
        await act(() => carosel.onProgressChange(1, 2));
        const paginationUpdate = tree.root.findByProps({
            testID: 'pagination'
        }).props;
        expect(paginationUpdate.activePage).toBe(2);
        const slideTree = renderer.create(<Slide index={2} />);
        const titleTextSlide2 = slideTree.root.findByProps({
            testID: 'titleTextSlide2'
        }).props;
        expect(titleTextSlide2).toBeTruthy();
    });
    it('scrolls to the next slide when "Next" button is pressed.', async () => {
        const useRefSpy = jest.spyOn(React, 'useRef');
        const carosel = tree.root.findByProps({
            testID: 'carosel'
        }).props;
        await act(() => carosel.onProgressChange(1, 0));
        const paginationUpdate = tree.root.findByProps({
            testID: 'pagination'
        }).props;
        expect(paginationUpdate.activePage).toBe(0);

        const nextButton = tree.root.findByProps({
            testID: 'nextButton'
        }).props;
        await act(() => nextButton.onPress());
        expect(useRefSpy).toHaveBeenCalled();
    });
    it('navigates to "Main" when "Let\'s Start" button is pressed on the last slide.', async () => {
        const carosel = tree.root.findByProps({
            testID: 'carosel'
        }).props;
        await act(() => carosel.onProgressChange(1, 2));
        const paginationUpdate = tree.root.findByProps({
            testID: 'pagination'
        }).props;
        expect(paginationUpdate.activePage).toBe(2);

        const startButton = tree.root.findByProps({
            testID: 'startButton'
        }).props;
        await act(() => startButton.onPress());
        expect(mockNavigate).toHaveBeenCalledWith('Main');
    });
    it('applies styles correctly.', async () => {
        const startButton = tree.root.findByProps({
            testID: 'startButton'
        }).props;
        expect(startButton.style).toStrictEqual({
            width: 300,
            height: 60,
            justifyContent: 'center',
            alignItems: 'center',
            backgroundColor: Colors.theme.primary,
            paddingHorizontal: 30,
            paddingVertical: 15,
            borderRadius: 10
        });
    });
});
