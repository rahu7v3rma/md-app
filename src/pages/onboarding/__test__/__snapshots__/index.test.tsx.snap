// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding snapshot onboarding 1`] = `
<View
  style={
    {
      "alignItems": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "height": "100%",
            "width": 750,
          },
          {},
        ]
      }
      testID="carosel"
    >
      <View
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          [
            {
              "flex": 1,
              "overflow": "hidden",
            },
            {
              "flexDirection": "column",
            },
            {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          collapsable={false}
          handlerTag={1}
          handlerType="PanGestureHandler"
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "height": "100%",
                "width": 750,
              },
              {},
              {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "height": "100%",
                  "position": "absolute",
                  "width": 750,
                },
                {
                  "transform": [
                    {
                      "translateX": undefined,
                    },
                  ],
                },
              ]
            }
          />
          <View
            style={
              [
                {
                  "height": "100%",
                  "position": "absolute",
                  "width": 750,
                },
                {
                  "transform": [
                    {
                      "translateX": undefined,
                    },
                  ],
                },
              ]
            }
          />
          <View
            style={
              [
                {
                  "height": "100%",
                  "position": "absolute",
                  "width": 750,
                },
                {
                  "transform": [
                    {
                      "translateX": undefined,
                    },
                  ],
                },
              ]
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "bottom": 20,
          "justifyContent": "space-between",
          "position": "absolute",
        }
      }
    >
      <View
        style={
          {
            "marginBottom": 45,
          }
        }
      >
        <View
          pointerEvents="box-none"
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingHorizontal": 20,
              "paddingVertical": 30,
            }
          }
          testID="pagination"
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={false}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "marginHorizontal": 8,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 2,
                    "height": 8,
                    "width": 8,
                  },
                  {
                    "backgroundColor": undefined,
                  },
                  {
                    "transform": [
                      {
                        "rotate": "undefineddeg",
                      },
                    ],
                  },
                ]
              }
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={false}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "marginHorizontal": 8,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 2,
                    "height": 8,
                    "width": 8,
                  },
                  {
                    "backgroundColor": undefined,
                  },
                  {
                    "transform": [
                      {
                        "rotate": "undefineddeg",
                      },
                    ],
                  },
                ]
              }
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={false}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "justifyContent": "center",
                "marginHorizontal": 8,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 2,
                    "height": 8,
                    "width": 8,
                  },
                  {
                    "backgroundColor": undefined,
                  },
                  {
                    "transform": [
                      {
                        "rotate": "undefineddeg",
                      },
                    ],
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#72996B",
            "borderRadius": 10,
            "height": 60,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 30,
            "paddingVertical": 15,
            "width": 300,
          }
        }
        testID="nextButton"
      >
        <Text
          adjustsFontSizeToFit={true}
          style={
            [
              {
                "fontFamily": "Poppins",
              },
              {
                "textAlign": "center",
              },
              {
                "color": "#FFFFFF",
                "fontSize": 14,
                "fontWeight": "600",
                "lineHeight": 22,
              },
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</View>
`;
